hub-detect/src/main/resources/hub-detect-4-sh

```diff
@@ -0,0 +1,104 @@
+# your own DETECT_JAR_PATH in your environment and
+# *that* location will be used.
+DETECT_JAR_PATH=${DETECT_JAR_PATH:-/tmp}
+
+# If you want to pass any java options to the
+# invocation, specify DETECT_JAVA_OPTS in your
+# environment. For example, to specify a 6 gigabyte
+# heap size, you would set DETECT_JAVA_OPTS=-Xmx6G.
+DETECT_JAVA_OPTS=${DETECT_JAVA_OPTS:-}
+
+# If you want to pass any additional options to
+# curl, specify DETECT_CURL_OPTS in your environment.
+# For example, to specify a proxy, you would set
+# DETECT_CURL_OPTS=--proxy http://myproxy:3128
+DETECT_CURL_OPTS=${DETECT_CURL_OPTS:-}
+
+SCRIPT_ARGS="$@"
+LOGGABLE_SCRIPT_ARGS=""
+
+for i in $*; do
+  if [[ $i == --blackduck.hub.password=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.hub.password=<redacted>"
+  elif [[ $i == --blackduck.hub.proxy.password=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.hub.proxy.password=<redacted>"
+  elif [[ $i == --blackduck.hub.api.token=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.hub.api.token=<redacted>"
+  elif [[ $i == --blackduck.password=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.password=<redacted>"
+  elif [[ $i == --blackduck.proxy.password=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.proxy.password=<redacted>"
+  elif [[ $i == --blackduck.api.token=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.api.token=<redacted>"
+  else
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS $i"
+  fi
+done
+
+run() {
+  get_detect
+  run_detect
+}
+
+get_detect() {
+  if [ -z "${DETECT_SOURCE}" ]; then
+    if [ -z "${DETECT_RELEASE_VERSION}" ]; then
+      VERSION_CURL_CMD="curl ${DETECT_CURL_OPTS} --silent --header \"X-Result-Detail: info\" 'https://test-repo.blackducksoftware.com/artifactory/api/storage/bds-integrations-release/com/blackducksoftware/integration/hub-detect?properties=${DETECT_VERSION_KEY}' | grep \"${DETECT_VERSION_KEY}\" | sed 's/[^[]*[^\"]*\"\([^\"]*\).*/\1/'"
+      DETECT_SOURCE=$(eval $VERSION_CURL_CMD)
+    else
+      DETECT_SOURCE="https://test-repo.blackducksoftware.com/artifactory/bds-integrations-release/com/blackducksoftware/integration/hub-detect/${DETECT_RELEASE_VERSION}/hub-detect-${DETECT_RELEASE_VERSION}.jar"
+    fi
+  fi
+  echo "will look for : ${DETECT_SOURCE}"
+
+  DETECT_FILENAME=${DETECT_FILENAME:-$(awk -F "/" '{print $NF}' <<< $DETECT_SOURCE)}
+  DETECT_DESTINATION="${DETECT_JAR_PATH}/${DETECT_FILENAME}"
+
+  USE_REMOTE=1
+  if [ ! -f $DETECT_DESTINATION ]; then
+    echo "You don't have the current file, so it will be downloaded."
+  else
+    echo "You have already downloaded the latest file, so the local file will be used."
+    USE_REMOTE=0
+  fi
+
+  if [ $USE_REMOTE -eq 1 ]; then
+    echo "getting ${DETECT_SOURCE} from remote"
+    curl $DETECT_CURL_OPTS --silent -L -o $DETECT_DESTINATION "${DETECT_SOURCE}"
```

Comment:
I'd recommend using -w "%{http_code}" for any curl commands, and then checking that that curl returns 200.

CURL TEST:
$ curl --silent -L -o /tmp/hub-detect-4.3.0.jar https://test-repo.blackducksoftware.com/artifactory/bds-integrations-release/com/blackducksoftware/integration/hub-detect/4.3.0/hub-detect-4.3.0.DANA
$ echo $?
0
$ cat /tmp/hub-detect-4.3.0.jar 
{
  "errors" : [ {
    "status" : 404,
    "message" : "File not found."
  } ]
}

$ curlReturn=$(curl --silent -L -o /tmp/hub-detect-4.3.0.jar -w "%{http_code}" https://test-repo.blackducksoftware.com/artifactory/bds-integrations-release/com/blackducksoftware/integration/hub-detect/4.3.0/hub-detect-4.3.0.DANA)
$ echo ${curlReturn} 
404

$ curlReturn=$(curl --silent -L -o /tmp/hub-detect-4.3.0.jar -w "%{http_code}" https://test-repo.blackducksoftware.com/artifactory/bds-integrations-release/com/blackducksoftware/integration/hub-detect/4.3.0/hub-detect-4.3.0.jar)
$ echo ${curlReturn} 
200

Reply:
I like this - we'll do this.